vendor/github.com/go-micro/plugins/v4/store/nats-js-kv/keys.go

```diff
@@ -79,17 +79,27 @@ func (k *Key) Check(table, prefix, suffix string) bool {
 	return true
 }
 
-func encode(s string) string {
-	return base32.StdEncoding.EncodeToString([]byte(s))
+func encode(s string, alg string) string {
+	switch alg {
+	case "base32":
+		return base32.StdEncoding.EncodeToString([]byte(s))
+	default:
+		return s
+	}
 }
 
-func decode(s string) string {
-	b, err := base32.StdEncoding.DecodeString(s)
-	if err != nil {
+func decode(s string, alg string) string {
+	switch alg {
+	case "base32":
+		b, err := base32.StdEncoding.DecodeString(s)
+		if err != nil {
+			return s
+		}
+
+		return string(b)
+	default:
```

Comment:
@kobergj ... so nats-js-kv used to _always_ encode the keys ... but doesn't that mean that we now also always have to encode the keys? which makes https://github.com/cs3org/reva/pull/4842#discussion_r1761003987 weird...